lib/rubocop/cop/style/ternary_parentheses.rb

```diff
@@ -203,16 +201,44 @@ def correct_parenthesized(corrector, condition)
           # If we remove the parentheses, we need to add a space or we'll
           # generate invalid code.
           corrector.insert_after(condition.loc.end, ' ') unless whitespace_after?(condition)
+
+          if (send_node = condition_send_node(condition)) && send_node_args_need_parens?(send_node)
+            parenthesize_condition_arguments(corrector, send_node)
+          end
         end
 
         def correct_unparenthesized(corrector, condition)
           corrector.wrap(condition, '(', ')')
         end
 
+        def parenthesize_condition_arguments(corrector, send_node)
+          range_start = send_node.defined_type? ? send_node.loc.keyword : send_node.loc.selector
+          opening_range = range_start.end.join(send_node.first_argument.source_range.begin)
+
+          corrector.replace(opening_range, '(')
+          corrector.insert_after(send_node.last_argument, ')')
+        end
+
         def whitespace_after?(node)
           last_token = processed_source.last_token_of(node)
           last_token.space_after?
         end
+
+        def condition_send_node(condition)
+          send_node = condition.child_nodes.last
+          send_node if node_with_args?(send_node)
```

Comment:
Apologies for the repetition feedback, it seems the tests pass even without the `if node_with_args?` guard condition.

Reply:
Yup true, because `send_node_args_need_parens?` calls `node_with_args?` too. I refactored away the `condition_send_node` method now.